src/Database/Query.php

```diff
@@ -889,6 +889,46 @@ public function where($conditions = null, $types = [], $overwrite = false)
         return $this;
     }
 
+    /**
+     * Adds an IN condition or set of conditions to be used in the WHERE clause for this
+     * query.
+     *
+     * This method does allow empty inputs in contrast to where().
+     * Be careful about using it without proper sanity checks.
+     *
+     * @param string $field Field
+     * @param array $values Array of values
+     * @return $this
+     */
+    public function whereIn($field, array $values)
+    {
+        if (!$values) {
```

Comment:
so `''`, `null`, `0` is not valid? How come?

`WHERE foo IN (1, 2, 3, NULL)` sounds reasonable to me so would `WHERE foo IN (NULL)`?

Reply:
They are all valid as long as they are passed as array - this uses the current internal behavior, so no special things here.

Reply:
So why is passing in `null` invalid, why is passing in `int(0)` invalid?

Reply:
@inoas Wouldn't the type hint prevent null/false/0 from entering the method?

Reply:
So I miss read the check and its meaning is?:

```php
if ($values === []) {
```

Can that be written then now that we got a strict type hint?

Reply:
Why? Same thing. Your comments are misleading to me.

Reply:
Well if it had been the `=== []` I'd not had miss-understood the code. Now to understand how the function runs you have to look at the signature and typehint.

Anyway, seems you like all the implicit weak castings. It works.